frontend/src/hooks/useResizeHandle.ts

```diff
@@ -80,7 +91,7 @@ export const useResizeHandle = ({
       document.removeEventListener("mousemove", handleMouseMove);
       document.removeEventListener("mouseup", handleMouseUp);
     };
-  }, [onResize]);
+  }, [maxWidth, minWidth, onResize]);
```

Comment:
I don't think this effect should rerun each time on the width changes, since they are adding and removing listeners. Instead maybe you could do 

`useEvent` to pull out the handlers, and then pass the handlers as deps here